packages/utilities/src/utilities/smoothScroll/smoothScroll.js

```diff
@@ -11,19 +11,27 @@
  * @example
  * import { smoothScroll } from '@carbon/ibmdotcom-utilities';
  *
- * <a href="#anchorlinkname" onClick={smoothScroll(e)}>lorem ipsum</a>
+ * Here e is event param and seletor is param where you want to apply smoothscroll
+ * <a href="#anchorlinkname" onClick={smoothScroll({ e, selector })}>lorem ipsum</a>
  *
  * You can use this for jump to target element by providing event object.
  * it will scroll into view of target by selecting attribute and assigning to id.
  *
- * @param {*} e event object
+ * @param {object} { e: Event, Selector: QuerySelector for the target element }
  */
-const smoothScroll = e => {
-  e.preventDefault();
-  const id = e.currentTarget.getAttribute('href');
-  document.querySelector(id).scrollIntoView({
-    behavior: 'smooth',
-    block: 'start',
-  });
+const smoothScroll = ({ e, selector }) => {
+  if (e !== null) {
+    e.preventDefault();
+    const id = e.currentTarget.getAttribute('href');
+    document.querySelector(id).scrollIntoView({
+      behavior: 'smooth',
+      block: 'start',
+    });
+  } else if (selector !== null) {
+    document.querySelector(selector).scrollIntoView({
+      behavior: 'smooth',
+      block: 'start',
+    });
+  }
```

Comment:
We can probably simplify this and reduce some of the redundancy down.

```javascript
const smoothScroll = ({ e, selector }) => {
  let getSelector;

  if (e !== null) {
    e.preventDefault();
    getSelector = e.currentTarget.getAttribute('href');
  } else if (selector !== null) {
    getSelector = selector;
  } else {
    return null;
  }

  document.querySelector(getSelector).scrollIntoView({
    behavior: 'smooth',
    block: 'start',
  });
};
```

Reply:
Okay